src/builtins/core/zoneddatetime.rs

```diff
@@ -555,16 +584,40 @@ impl ZonedDateTime {
 impl ZonedDateTime {
     /// Creates a new valid `ZonedDateTime`.
     #[inline]
-    pub fn try_new(nanos: i128, calendar: Calendar, time_zone: TimeZone) -> TemporalResult<Self> {
+    pub fn try_new_with_provider(
+        nanos: i128,
+        calendar: Calendar,
+        time_zone: TimeZone,
+        provider: &impl TimeZoneProvider,
+    ) -> TemporalResult<Self> {
         let instant = Instant::try_new(nanos)?;
-        Ok(Self::new_unchecked(instant, calendar, time_zone))
+        Self::new_unchecked_with_provider(instant, calendar, time_zone, provider)
+    }
+    /// Creates a new valid `ZonedDateTime`.
+    #[inline]
+    pub fn try_new(
+        nanos: i128,
+        calendar: Calendar,
+        time_zone: TimeZone,
+        cached_offset: UtcOffset,
```

Comment:
issue: I'm not sold that the cached_offset should be leaking into the public API.

Couldn't this also cause other issues, like someone providing `America/Los_Angeles` and a +3600 UTC offset?

Reply:
I'd definitely prefer that in this case.